GameServerLib/Logic/GameObjects/Unit.cs

```diff
@@ -739,6 +744,36 @@ public ClassifyUnit ClassifyTarget(Unit target)
             return ClassifyUnit.Default;
         }
 
+        public byte GetBuffSlot(Buff b)
+        {
+            foreach (var buff in BuffSlotToRemove)
+            {
+                if (!AppliedBuffs.Values.Contains(buff))
+                    continue;
+
+                for (byte i = 1; i <= AppliedBuffs.Count + 1; i++)
+                {
+                    if (!AppliedBuffs.ContainsKey(i))
+                        continue;
+                    if (AppliedBuffs[i].Equals(buff))
+                        AppliedBuffs.Remove(i);
+                }
+            }
+            for (byte i = 1; i <= AppliedBuffs.Count + 1; i++)
+            {
+                if (AppliedBuffs.ContainsKey(i))
+                    continue;
+                AppliedBuffs.Add(i, b);
```

Comment:
Why are we updating buff slots in a function called `GetBuffSlot`?

At the very least make the function name imply what it does, `UpdateBuffSlots` for example. Also would it not make more sense to remove the buffs that you want to remove in the `RemoveBuffSlot` function instead of here? Do we really need a `BuffSlotToRemove` variable?